### PR TITLE
CEO PR# 5.1 - CEO Group 1 - Bjorn, Naomi, Ulrich

### PR DESCRIPTION
**Pull request by d-little**
### CEO PR5 - Small sets of CEOs

PR5.1 has Bjorn, Naomi, Ulrich.

### Ulrich

Pretty straightfoward and self contained.  Hopefully fully covered with testing

### Bjorn

I forgot at first that this will affect both Warmonger award and Law Suit.  

A lot of test coverage for this card.  But I'm not sure how to implement tests for Warmonger Award?  I'm not sure there actually _are_ tests for this award

### Naomi

Small hook when building a Colony.

The implementation itself _could_ be better in terms of UI.  It currently runs through every active colony with an increase/decrease prompt.  It's very clear what choices you're making though, so I'll leave it as-is

### CODE REVIEW

File: src/server/colonies/Colony.ts [hunk 1]
```diff
@@ -92,6 +92,12 @@ export abstract class Colony implements IColony {
       if (poseidon !== undefined) {
         poseidon.production.add(Resources.MEGACREDITS, 1);
       }
+
+      // CEO Naomi hook
+      if (player.cardIsInEffect(CardName.NAOMI)) {
```
Comment (kberg) [inline src/server/colonies/Colony.ts:97]: Looks like it's time for an onColonyAdded method on cards. Doesn't have to be now.

**Review by kberg — COMMENTED** [review]
Basically looks great! I agree, there are probably no Warmonger tests. Don't worry about it.

You said something about Law Suit -- is that handled?

File: src/server/cards/leaders/Bjorn.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  public canAct(): boolean {
+    return this.isDisabled === false;
+  }
+
+  public action(player: Player): PlayerInput | undefined {
+    const game = player.game;
+    const targetPlayers = game.getPlayers().filter((p) => p.id !== player.id && p.megaCredits > player.megaCredits);
+
+    targetPlayers.forEach((target) => {
+      const qtyToSteal = Math.min(game.generation, target.megaCredits);
```
[the remark this answers is not in the document]
Comment (d-little) [reply src/server/cards/leaders/Bjorn.ts:43]: > There's a method in Player that simplifies this. (`stealResource`)

Ah! I think I had problems with this originally, and now I know why:  I misread the name 'from:' to mean 'steal _from_ this player, and give to whoever called this' and all my tests kept failing.  I got it now.

[the remark this answers is not in the document]
Comment (d-little) [reply src/server/cards/leaders/Bjorn.ts:43]: > You said something about Law Suit -- is that handled?

It's confirmed working as intended in tests, I just needed to make sure it was working.  No changes needed in other files.

File: src/server/cards/leaders/Ulrich.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  public canAct(): boolean {
+    return this.isDisabled === false;
+  }
+
+  public action(player: Player): PlayerInput | undefined {
+    const game = player.game;
+    const oceansPlaced = game.board.getOceanCount();
+    let bonus_credits = 15; // Default to 15, this happens if all oceans have been placed
```
[the remark this answers is not in the document]
Comment (d-little) [reply src/server/cards/leaders/Ulrich.ts]: > s/bonus_credits/bonusCredits. This is also minor.

The python in me keeps coming through.   Simplified version is neat, I'll need to try to remember that I can do that in TS too

**Pull request merged**